components/support/sync-telemetry/src/main/java/mozilla/components/support/sync/telemetry/SyncTelemetry.kt

```diff
@@ -156,44 +159,126 @@ object SyncTelemetry {
-                    }
-                    if (base.reconciled > 0) {
-                        incoming["reconciled"].add(base.reconciled)
-                    }
-                    if (base.uploaded > 0) {
-                        outgoing["uploaded"].add(base.uploaded)
-                    }
-                    if (base.failedToUpload > 0) {
-                        outgoing["failed_to_upload"].add(base.failedToUpload)
-                    }
-                    if (base.outgoingBatches > 0) {
-                        outgoingBatches.add(base.outgoingBatches)
-                    }
-                    base.failureReason?.let {
-                        recordFailureReason(it, failureReason)
-                    }
-                    engine.validation?.let {
-                        it.problems.forEach {
-                            remoteTreeProblems[it.name].add(it.count)
-                        }
-                    }
-                }
+                individualBookmarksSync(ping.uid, engine)
                 sendPing()
             }
         }
         return true
     }
 
+    @Suppress("ComplexMethod")
+    private fun individualLoginsSync(hashedFxaUid: String, engineInfo: EngineInfo, syncUuid: UUID? = null) {
+        require(engineInfo.name == "passwords") { "Expected 'passwords', got ${engineInfo.name}" }
+
+        syncUuid?.let { Sync.syncUuid.set(syncUuid) }
+        LoginsSync.apply {
+            val base = BaseGleanSyncPing.fromEngineInfo(hashedFxaUid, engineInfo)
+            uid.set(base.uid)
+            startedAt.set(base.startedAt)
+            finishedAt.set(base.finishedAt)
+            if (base.applied > 0) {
+                // Since all Sync ping counters have `lifetime: ping`, and
+                // we send the ping immediately after, we don't need to
+                // reset the counters before calling `add`.
+                incoming["applied"].add(base.applied)
+            }
+            if (base.failedToApply > 0) {
+                incoming["failed_to_apply"].add(base.failedToApply)
+            }
+            if (base.reconciled > 0) {
+                incoming["reconciled"].add(base.reconciled)
+            }
+            if (base.uploaded > 0) {
+                outgoing["uploaded"].add(base.uploaded)
+            }
+            if (base.failedToUpload > 0) {
+                outgoing["failed_to_upload"].add(base.failedToUpload)
+            }
+            if (base.outgoingBatches > 0) {
+                outgoingBatches.add(base.outgoingBatches)
+            }
+            base.failureReason?.let {
+                recordFailureReason(it, failureReason)
+            }
+        }
+    }
+
+    @Suppress("ComplexMethod")
+    private fun individualBookmarksSync(hashedFxaUid: String, engineInfo: EngineInfo, syncUuid: UUID? = null) {
+        require(engineInfo.name == "bookmarks") { "Expected 'bookmarks', got ${engineInfo.name}" }
+
+        syncUuid?.let { Sync.syncUuid.set(syncUuid) }
```

Comment:
I don't think this line is required: you are already setting the UUID and, because you are saying in `pings.yaml` it should go in all the pings, then it will go in all the pings.